specs/commandline/inspect.md

```diff
@@ -0,0 +1,271 @@
+# notation inspect
+
+## Description
+
+Use `notation inspect` command to inspect all the signatures associated with signed artifact in a human readable format.
+
+Upon successful execution, both the digest of the signed artifact and the digests of signatures manifest along with their properties associated with the signed artifact are printed in the following format:
+
+```shell
+<registry>/<repository>@<digest>
+└──application/vnd.cncf.notary.signature
+  ├──<digest_of_signature_manifest>
+    ├──<signingAlgorithm>
+    ├──<signedAttributes>
+    ├──<userDefinedAttributes>
+    ├──<unsignedAttributes>
+    ├──<certificates>
+    └──<payload>
+  ├──<digest_of_signature_manifest>
+    ├──<signingAlgorithm>
+    ├──<signedAttributes>
+    ├──<unsignedAttributes>
+    ├──<certificates>
+    └──<payload>
+```
+
+## Outline
+
+```text
+Inspect all signatures with the signed artifact.
+
+Usage:
+    notation inspect [flags] <reference>
+  
+Flags:
+   -h, --help              help for describing the signature
+   -o, --output json       output on command line sets the output to json
```

Comment:
Consider `--display {text,tree,json}` over `--output` since `--output` is common as output path.

Reference: https://clig.dev/#output
> `-o`, `--output`: Output file. For example, `sort`, `gcc`.

/cc @FeynmanZhou @yizha1 for `oras discover`.

Reply:
Originally, https://github.com/notaryproject/notation/pull/490#discussion_r1062147121 by @yizha1 

> Since we need to add this kind of `output` flag for other notation cli commands, and it is on the user interface, it's better we can align what is the best solution for our users. Here are 3 options so far:
> 
> 1. `-o, --output json`
> 2. `--display {tree, json) by default tree
> 3. --json (an example in https://clig.dev/#output)
> 
> Any other options, Let's vote?

Reply:
> Originally, [#490 (comment)](https://github.com/notaryproject/notation/pull/490#discussion_r1062147121) by @yizha1
> 
> > Since we need to add this kind of `output` flag for other notation cli commands, and it is on the user interface, it's better we can align what is the best solution for our users. Here are 3 options so far:
> > 
> > 1. `-o, --output json`
> > 2. `--display {tree, json) by default tree
> > 3. --json (an example in https://clig.dev/#output)
> > 
> > Any other options, Let's vote?

No changes to the output command as per the discussions and alignment in the community meeting as of 01/09/2023